### PR TITLE
Add validation and ghe

### PR DESCRIPTION
**Pull request by petro2061**
Added valigation package. Packege include custom annotation and implementation for validation input data to server. To pon.xml fille add new dependency and added special annotation on controller methots. Also imlemented CustomExsertionHandler class. This class include methods for handling exceptions at the application level.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -90,7 +90,10 @@
 			<version>${liquibase.version}</version>
 		</dependency>
 
-
+		<dependency>
+			<groupId>org.hibernate.validator</groupId>
```
Comment (Elena-Bruyako) [inline pom.xml]: It's better to use spring-boot-starter-validation which provides hibernate-validator also

File: src/main/java/project/onlinebookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
+import project.onlinebookstore.validation.Author;
+import project.onlinebookstore.validation.CoverImage;
+import project.onlinebookstore.validation.Isbn;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull(message = "Field title can't be null")
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
    @NotBlank(message = "Field title can't be null")
```

File: src/main/java/project/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            DataProcessingException ex) {
+        return new ResponseEntity<>(getBody(ex), HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<Map<String, Object>> handleAllExceptions(Exception ex) {
+        return new ResponseEntity<>(getBody(ex), HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler(NullPointerException.class)
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: remove

File: src/main/java/project/onlinebookstore/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
     private String title;
+    @Author
     private String author;
+    @Isbn
     private String isbn;
+    @NotNull(message = "Field price can't be null")
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
    @NotBlank(message = "Field price can't be null")
```

File: src/main/java/project/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    public ResponseEntity<Map<String, Object>> handleNullPointerExceptions(
+            NullPointerException ex) {
+        return new ResponseEntity<>(getBody(ex), HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler(NoSuchElementException.class)
+    public ResponseEntity<Map<String, Object>> handleNoSuchElementException(
+            NullPointerException ex) {
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: First, why NullPointerException? 
Second, in which places you catching NoSuchElementException?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's improve your solution

**Comment by petro2061** [comment]
Added another dependency to pom.xml. Changed the annotation from @NotNull to @NotBlank and removed redundant methods.

File: src/main/java/project/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    @ExceptionHandler(DataProcessingException.class)
+    public ResponseEntity<Map<String, Object>> handleDataProcessingException(
+            DataProcessingException ex) {
+        return new ResponseEntity<>(getBody(ex), HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<Map<String, Object>> handleEntityNotFoundException(
+            EntityNotFoundException ex) {
+        return new ResponseEntity<>(getBody(ex), HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(getBody(ex), HttpStatus.NOT_FOUND);
```

File: src/main/java/project/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                .toList();
+
+        body.put("timestamp", LocalDateTime.now());
+        body.put("code", HttpStatus.BAD_REQUEST.value());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(DataProcessingException.class)
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: Did you use this exception? 
If so, please fix the HttpStatus

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Comment by petro2061** [comment]
Changed HttpStatus and deleted not use handler

**Review by Elena-Bruyako — APPROVED** [review]
Good job!

**Comment by Elena-Bruyako** [comment]
Rejected my mistake on the platform. Could you please send PR again?

**Pull request merged**